### PR TITLE
change tekton pipeline metadata to generatename

### LINKED ISSUE #97: make each pipeline run name unique
The `PipelineRun`s triggered by the CNBi controller are named after their corresponding `cnbi` object name and buildType.

Two `PipelineRun`s from the same object would have a name clash.

Each `PipelineRun` should have a unique name to prevent such a clash.

## Acceptance criteria

- [ ] each pipeline run created from the meteor operator has a unique name (possibly based on the build type, but with extra unique ids)

### PR DESCRIPTION
**Pull request by KPostOffice**
Signed-off-by: Kevin <kpostlet@redhat.com>

## Related Issues and Dependencies
fixes #97 

## This introduces a breaking change

- Yes
- No

## This should yield a new module release

- Yes
- No

## This Pull Request implements
This is just changing to using generate name for the PipelineRun resource. I'm unsure what other changes need to be done wrt the rest of the reconciliation loop.

### CODE REVIEW

File: controllers/cre/controller.go [hunk 1]
```diff
@@ -137,6 +137,7 @@ func (r *CustomRuntimeEnvironmentReconciler) reconcilePipelineRun(ctx context.Co
 		if errors.IsNotFound(err) {
 			logger.Info("Creating PipelineRun")
 
+			// TODO: replace ArrayOrString (deprecated) https://github.com/tektoncd/pipeline/blob/75d0037daf06664cb93efe86d974e1fbf69cbbbe/pkg/apis/pipeline/v1beta1/param_types.go#L140
```
Comment (goern) [inline controllers/cre/controller.go:140]: @KPostOffice could you relate the tekton upstream version to an OpenShift product version?

File: controllers/cre/controller.go [hunk 2]
```diff
@@ -109,8 +109,8 @@ func (r *CustomRuntimeEnvironmentReconciler) reconcilePipelineRun(ctx context.Co
 	pipeline := build_types[cre.Spec.BuildType]
 	pipelineRun := &pipelinev1beta1.PipelineRun{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("cre-%s-%s", cre.GetName(), pipeline),
-			Namespace: cre.Namespace,
+			GenerateName: fmt.Sprintf("cre-%s-%s-", cre.GetName(), pipeline),
+			Namespace:    cre.Namespace,
 			Labels: map[string]string{
 				"cre.thoth-station.ninja/pipeline": build_types[cre.Spec.BuildType],
```
Comment (goern) [inline controllers/cre/controller.go:115]: @KPostOffice could you add a label refering to the CRE generation that created the PipelineRun

**Review by goern — CHANGES_REQUESTED** [review]

File: controllers/cre/controller.go [hunk 3]
```diff
@@ -109,10 +110,11 @@ func (r *CustomRuntimeEnvironmentReconciler) reconcilePipelineRun(ctx context.Co
 	pipeline := build_types[cre.Spec.BuildType]
 	pipelineRun := &pipelinev1beta1.PipelineRun{
 		ObjectMeta: metav1.ObjectMeta{
-			Name:      fmt.Sprintf("cre-%s-%s", cre.GetName(), pipeline),
-			Namespace: cre.Namespace,
+			GenerateName: fmt.Sprintf("cre-%s-%s-", cre.GetName(), pipeline),
```
Comment (codificat) [inline controllers/cre/controller.go:113]: We should check if a PipelineRun already exists for this CRE

**Review by codificat — CHANGES_REQUESTED** [review]
There's a problem here: this will get a new PipelineRun name every time the reconcile function is called - without checking if there's an existing PR for the CRE.

This results in a storm of PipelineRuns being created for every CRE.

File: controllers/cre/controller.go (hunk 3 above)
Comment (KPostOffice) [reply controllers/cre/controller.go:113]: Would it be okay to to add the generation directly to the name? Like:
```
Name: fmt.Sprintf("cre-%s-%s-%s", cre.GetName(), cre.GetGeneration(), pipeline)
```

Comment (goern) [reply controllers/cre/controller.go:113]: Let's conclude discussions in #136

Comment (VannTen) [reply controllers/cre/controller.go:113]: @KPostOffice generation in the pipeline name can't hurt, but we would also need a label with the generation if we choose that scheme, so we can query with label the K8S api.

File: controllers/cre/controller.go (hunk 1 above)
Comment (KPostOffice) [reply controllers/cre/controller.go:142]: The comment exists from 0.39.0 onward. I found [this](https://docs.openshift.com/container-platform/4.11/cicd/pipelines/op-release-notes.html#compatibility-support-matrix_op-release-notes) compatibility matrix. It states that operator version 1.8, which is the latest pipeline operator version, is tekton version 0.37.x.

The version we are currently using in this repo is v0.38.3. This isn't an issue for now.

**Comment by codificat** [comment]
Tested in a test cluster and looking good, thanks!

/approve

**Pull request merged**